src/app/notifications/components/UpdatesAvailable.tsx

```diff
@@ -16,16 +16,17 @@ import { Body, Group, Header, P } from './Card';
 
 // tslint:disable-next-line:variable-name
 const UpdatesAvailable = ({className}: {className?: string}) => {
-  const sw = useServices().serviceWorker;
-  const [readyToReload, setReadyToReload] = useState<boolean>(!serviceWorkerNeedsUpdate(sw));
+  const serviceWorkerContainer = assertWindow().navigator.serviceWorker;
+  const [sw, setSw] = useState<ServiceWorkerRegistration | undefined>(undefined);
 
-  useEffect(() => {
-    findAndInstallServiceWorkerUpdate(sw, () => setReadyToReload(true));
-  // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  if (serviceWorkerContainer) {
+    serviceWorkerContainer.ready.then((registration) => {
```

Comment:
you shouldn't be binding a promise handler outside of useEffect, or it'll be re-bound every time the component renders

Reply:
Added useEffect()